File: AntiClimacus/AntiClimacus/Models/DonorBloodDonation.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace AntiClimacus.Models
{
    public class DonorBloodDonation
    {
        public int DonorId { get; set; }
        public List<BloodComponentQuantity> BloodComponents { get; set; }
        public bool Diseases { get; set; }

        public DonorBloodDonation()
        {

        }
    }
}

File: DonatorSange-Angular/src/app/models/donorBloodDonation.ts
import { BloodComponentQuantity } from "./bloodComponentQuantity";

export class DonorBloodDonation{
    donorId: number;
    bloodComponents: Array<BloodComponentQuantity>;
    diseases: boolean;

    constructor(id:number, components:Array<BloodComponentQuantity>, diseases:boolean){
        this.donorId=id;
        this.bloodComponents=components;
        this.diseases=diseases;
    }
}